### PR TITLE
Add sanity check into SemanticTest for not allowing ABIEncoderV1Only and compileViaYul together

### PR DESCRIPTION
**Pull request by axic**
Slimmed down version of #10643.

### CODE REVIEW

File: test/libsolidity/semanticTests/revertStrings/empty_v1_yul.sol [hunk 1]
```diff
@@ -1,18 +0,0 @@
-contract C {
```
Comment (axic) [inline test/libsolidity/semanticTests/revertStrings/empty_v1_yul.sol:1]: This test is a duplicate of `empty_v1.sol` with the weird set of conditions: ABIEncoderV1Only: true andcompileViaYul: true

File: test/libsolidity/SemanticTest.cpp [hunk 2]
```diff
@@ -99,6 +99,10 @@ SemanticTest::SemanticTest(
 	if (m_runWithABIEncoderV1Only && !solidity::test::CommonOptions::get().useABIEncoderV1)
 		m_shouldRun = false;
 
+	// Sanity check
+	if (m_runWithABIEncoderV1Only && (compileViaYul == "true" || compileViaYul == "also"))
+		BOOST_THROW_EXCEPTION(runtime_error("ABIEncoderV1Only can not be used with compileViaYul=" + compileViaYul + ", set it to false or omit the flag."));
```
Comment (cameel) [inline test/libsolidity/SemanticTest.cpp]: Might be a good idea to say why. I.e. that `abicoder v1` pragma is just ignored in Yul mode and everything actually runs as if v2 was used.

File: test/libsolidity/semanticTests/arithmetics/checked_add_v2.sol [hunk 3]
```diff
@@ -0,0 +1,14 @@
+pragma abicoder v2;
+contract C {
+    // Input is still not checked - this needs ABIEncoderV2!
```
Comment (cameel) [inline test/libsolidity/semanticTests/arithmetics/checked_add_v2.sol]: Is this comment a left-over from something older? This test does have the v2 pragma.

**Review by cameel — DISMISSED** [review]
Looks fine.

Just some comments may need tweaking.

File: test/libsolidity/semanticTests/arithmetics/checked_add_v2.sol (hunk 3 above)
Comment (axic) [reply test/libsolidity/semanticTests/arithmetics/checked_add_v2.sol]: It is a copy of the `checked_add_v1.sol` file.

Comment (axic) [reply test/libsolidity/semanticTests/arithmetics/checked_add_v2.sol]: Fixed.

File: test/libsolidity/SemanticTest.cpp (hunk 2 above)
Comment (axic) [reply test/libsolidity/SemanticTest.cpp]: Fixed.

**Review by cameel — APPROVED** [review]

**Pull request merged**